tensorzero-internal/tests/e2e/providers/common.rs

```diff
@@ -548,31 +552,31 @@ pub async fn test_image_inference_with_provider_filesystem(provider: E2ETestProv
     assert_eq!(result, FERRIS_PNG);
 }
 
-#[cfg_attr(not(feature = "e2e_tests"), allow(dead_code))]
+#[cfg(feature = "e2e_tests")]
 pub async fn test_image_inference_with_provider_s3_compatible(provider: E2ETestProvider) {
+    use rand::distributions::Alphanumeric;
+    use rand::distributions::DistString;
```

Comment:
why is this in the function?

Reply:
This is only used inside a `#[cfg(feature = "e2e_tests")]` function, so a top-level import will be unused without that feature.